axon-mongo-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/mongo/springboot/AxonMongoProperties.java

```diff
@@ -0,0 +1,203 @@
+     */
+    private String databaseName = null;
+
+    /**
+     * The AutoConfiguration settings for the token store
+     */
+    private TokenStore tokenStore = new TokenStore();
+
+    /**
+     * The AutoConfiguration settings for the event store
+     */
+    private EventStore eventStore = new EventStore();
+
+    /**
+     * The AutoConfiguration settings for the saga store
+     */
+    private SagaStore sagaStore = new SagaStore();
+
+    /**
+     * Retrieves the database name to use for the Axon Framework Mongo Collections, will default to the one the
+     * {@link org.springframework.data.mongodb.MongoDatabaseFactory} used by default.
+     *
+     * @return the database name to used for the Axon Framework Mongo Collections
+     */
+    public String getDatabaseName() {
+        return databaseName;
+    }
+
+    /**
+     * Sets the database name to use for the Axon Framework Mongo Collections, will default to the one the
+     * {@link org.springframework.data.mongodb.MongoDatabaseFactory} used by default.
+     *
+     * @param databaseName the database name to use for the Axon Framework Mongo Collections
+     */
+    public void setDatabaseName(String databaseName) {
+        this.databaseName = databaseName;
+    }
+
+    /**
+     * Retrieves the AutoConfiguration settings for the token store.
+     *
+     * @return the AutoConfiguration settings for the token store
+     */
+    public TokenStore getTokenStore() {
+        return tokenStore;
+    }
+
+    /**
+     * Defines the AutoConfiguration settings for the token store.
+     *
+     * @param tokenStore the AutoConfiguration settings for the token store.
+     */
+    public void setTokenStore(TokenStore tokenStore) {
+        this.tokenStore = tokenStore;
+    }
+
+    /**
+     * Retrieves the AutoConfiguration settings for the event store.
+     *
+     * @return the AutoConfiguration settings for the event store
+     */
+    public EventStore getEventStore() {
+        return eventStore;
+    }
+
+    /**
+     * Defines the AutoConfiguration settings for the event store.
+     *
+     * @param eventStore the AutoConfiguration settings for the event store.
+     */
+    public void setEventStore(EventStore eventStore) {
+        this.eventStore = eventStore;
+    }
+
+    /**
+     * Retrieves the AutoConfiguration settings for the saga store.
+     *
+     * @return the AutoConfiguration settings for the saga store
+     */
+    public SagaStore getSagaStore() {
+        return sagaStore;
+    }
+
+    /**
+     * Defines the AutoConfiguration settings for the saga store.
+     *
+     * @param sagaStore the AutoConfiguration settings for the saga store.
+     */
+    public void setSagaStore(SagaStore sagaStore) {
+        this.sagaStore = sagaStore;
+    }
+
+    public static class TokenStore {
+
+        /**
+         * Enables creation of the {@link org.axonframework.extensions.mongo.eventsourcing.tokenstore.MongoTokenStore}
```

Comment:
I think it's good to state the default here. That will show up in user's IDE in a `*.properties` file.

Reply:
added